jib-core/src/integration-test/java/com/google/cloud/tools/jib/api/ReproducibleImageTest.java

```diff
@@ -89,6 +94,50 @@ public static void createImage()
         .containerize(containerizer);
   }
 
+  @Test
+  public void testTarBallStructure() throws IOException {
+    // known content should produce known results
+    Iterator<String> fileNames =
+        Iterators.forArray(
+            "c46572ef74f58d95e44dd36c1fbdfebd3752e8b56a794a13c11cfed35a1a6e1c.tar.gz",
+            "6d2763b0f3940d324ea6b55386429e5b173899608abf7d1bff62e25dd2e4dcea.tar.gz",
+            "530c1954a2b087d0b989895ea56435c9dc739a973f2d2b6cb9bb98e55bbea7ac.tar.gz",
+            "config.json",
+            "manifest.json");
+    try (TarArchiveInputStream input = new TarArchiveInputStream(new FileInputStream(imageTar))) {
+      TarArchiveEntry imageEntry;
+      while ((imageEntry = input.getNextTarEntry()) != null) {
+        Assert.assertTrue(fileNames.hasNext());
+        Assert.assertEquals(fileNames.next(), imageEntry.getName());
+      }
+      Assert.assertFalse("more files expected: " + fileNames, fileNames.hasNext());
+    }
+  }
+
+  @Test
+  public void testManifest() throws IOException {
+    try (FileInputStream input = new FileInputStream(imageTar)) {
```

Comment:
Is there a reason to create `FileInputStream()` directly? We normally do `Files.newInputStream()`.

Reply:
Oh yeah, I had forgotten about GoogleCloudPlatform/google-cloud-eclipse#1898